### PR TITLE
influxd Configuration overhaul

### LINKED ISSUE #1726: influxd config options page not up to date
https://docs.influxdata.com/influxdb/v2.0/reference/cli/influxd/

I don't see any of the tls settings, but there might be othere.

```
➜  influxdb git:(master) ✗ ./bin/darwin/influxd -h

	Start up the daemon configured with flags/env vars/config file.

	The order of precedence for config options are as follows (1 highest, 3 lowest):
		1. flags
		2. env vars
		3. config file

	A config file can be provided via the INFLUXD_CONFIG_PATH env var. If a file is
	not provided via an env var, influxd will look in the current directory for a
	config.yaml file. If one does not exist, then it will continue unchanged.

Usage:
  influxd [flags]
  influxd [command]

Available Commands:
  help        Help about any command
  inspect     Commands for inspecting on-disk database data
  run         Start the influxd server (default)
  upgrade     Upgrade a 1.x version of InfluxDB
  version     Print the influxd server version

Flags:
      --assets-path string                                            override default assets by serving from a specific directory (developer mode)
      --bolt-path string                                              path to boltdb database (default "/Users/rsavage/.influxdbv2/influxd.bolt")
      --e2e-testing                                                   add /debug/flush endpoint to clear stores; used for end-to-end tests
      --engine-path string                                            path to persistent engine files (default "/Users/rsavage/.influxdbv2/engine")
      --feature-flags stringToString                                  feature flag overrides (default [])
  -h, --help                                                          help for influxd
      --http-bind-address string                                      bind address for the REST HTTP API (default ":8086")
      --influxql-max-select-buckets int                               The maximum number of group by time bucket a SELECT can create. A value of zero will max the maximum number of buckets unlimited.
      --influxql-max-select-point int                                 The maximum number of points a SELECT can process. A value of 0 will make the maximum point count unlimited. This will only be checked every second so queries will not be aborted immediately when hitting the limit.
      --influxql-max-select-series int                                The maximum number of series a SELECT can run. A value of 0 will make the maximum series count unlimited.
      --log-level string                                              supported log levels are debug, info, and error (default "info")
      --no-tasks                                                      disables the task scheduler
      --query-concurrency int                                         the number of queries that are allowed to execute concurrently (default 10)
      --query-initial-memory-bytes int                                the initial number of bytes allocated for a query when it is started. If this is unset, then query-memory-bytes will be used
      --query-max-memory-bytes int                                    the maximum amount of memory used for queries. If this is unset, then this number is query-concurrency * query-memory-bytes
      --query-memory-bytes int                                        maximum number of bytes a query is allowed to use at any given time. This must be greater or equal to query-initial-memory-bytes (default 9223372036854775807)
      --query-queue-size int                                          the number of queries that are allowed to be awaiting execution before new queries are rejected (default 10)
      --reporting-disabled                                            disable sending telemetry data to https://telemetry.influxdata.com every 8 hours
      --secret-store string                                           data store for secrets (bolt or vault) (default "bolt")
      --session-length int                                            ttl in minutes for newly created sessions (default 60)
      --session-renew-disabled                                        disables automatically extending session ttl on request
      --storage-cache-max-memory-size Size                            The maximum size a shard's cache can reach before it starts rejecting writes. (default 1.0 GiB)
      --storage-cache-snapshot-memory-size Size                       The size at which the engine will snapshot the cache and write it to a TSM file, freeing up memory. (default 25 MiB)
      --storage-cache-snapshot-write-cold-duration Duration           The length of time at which the engine will snapshot the cache and write it to a new TSM file if the shard hasn't received writes or deletes. (default 10m0s)
      --storage-compact-full-write-cold-duration Duration             The duration at which the engine will compact all TSM files in a shard if it hasn't received a write or delete. (default 4h0m0s)
      --storage-compact-throughput-burst Size                         The rate limit in bytes per second that we will allow TSM compactions to write to disk. (default 48 MiB)
      --storage-max-concurrent-compactions int                        The maximum number of concurrent full and level compactions that can run at one time.  A value of 0 results in 50% of runtime.GOMAXPROCS(0) used at runtime.  Any number greater than 0 limits compactions to that value.  This setting does not apply to cache snapshotting.
      --storage-max-index-log-file-size Size                          The threshold, in bytes, when an index write-ahead log file will compact into an index file. Lower sizes will cause log files to be compacted more quickly and result in lower heap usage at the expense of write throughput. (default 1.0 MiB)
      --storage-retention-check-interval Duration                     The interval of time when retention policy enforcement checks run. (default 30m0s)
      --storage-series-file-max-concurrent-snapshot-compactions int   The maximum number of concurrent snapshot compactions that can be running at one time across all series partitions in a database.
      --storage-series-id-set-cache-size int                          The size of the internal cache used in the TSI index to store previously calculated series results.
      --storage-shard-precreator-advance-period Duration              The default period ahead of the endtime of a shard group that its successor group is created. (default 30m0s)
      --storage-shard-precreator-check-interval Duration              The interval of time when the check to pre-create new shards runs. (default 10m0s)
      --storage-tsm-use-madv-willneed                                 Controls whether we hint to the kernel that we intend to page in mmap'd sections of TSM files.
      --storage-validate-keys                                         Validates incoming writes to ensure keys only have valid unicode characters.
      --storage-wal-fsync-delay Duration                              The amount of time that a write will wait before fsyncing. A duration greater than 0 can be used to batch up multiple fsync calls. This is useful for slower disks or when WAL write contention is seen. (default 0s)
      --store string                                                  backing store for REST resources (bolt or memory) (default "bolt")
      --testing-always-allow-setup                                    ensures the /api/v2/setup endpoint always returns true to allow onboarding
      --tls-cert string                                               TLS certificate for HTTPs
      --tls-key string                                                TLS key for HTTPs
      --tls-min-version string                                        Minimum accepted TLS version (default "1.2")
      --tls-strict-ciphers                                            Restrict accept ciphers to: ECDHE_RSA_WITH_AES_256_GCM_SHA384, ECDHE_RSA_WITH_AES_256_CBC_SHA, RSA_WITH_AES_256_GCM_SHA384, RSA_WITH_AES_256_CBC_SHA
      --tracing-type string                                           supported tracing types are log, jaeger
      --vault-addr string                                             address of the Vault server expressed as a URL and port, for example: https://127.0.0.1:8200/.
      --vault-cacert string                                           path to a PEM-encoded CA certificate file on the local disk. This file is used to verify the Vault server's SSL certificate. This environment variable takes precedence over VAULT_CAPATH.
      --vault-capath string                                           path to a directory of PEM-encoded CA certificate files on the local disk. These certificates are used to verify the Vault server's SSL certificate.
      --vault-client-cert string                                      path to a PEM-encoded client certificate on the local disk. This file is used for TLS communication with the Vault server.
      --vault-client-key string                                       path to an unencrypted, PEM-encoded private key on disk which corresponds to the matching client certificate.
      --vault-client-timeout duration                                 timeout variable. The default value is 60s.
      --vault-max-retries int                                         maximum number of retries when a 5xx error code is encountered. The default is 2, for three total attempts. Set this to 0 or less to disable retrying.
      --vault-skip-verify                                             do not verify Vault's presented certificate before communicating with it. Setting this variable is not recommended and voids Vault's security model.
      --vault-tls-server-name string                                  name to use as the SNI host when connecting via TLS.
      --vault-token string                                            vault authentication token

Use "influxd [command] --help" for more information about a command.
```

### LINKED ISSUE #1727: Fix table formatting
Update Flag table on this page https://docs.influxdata.com/influxdb/v2.0/reference/cli/influxd/

refers to Layouts\shortcodes\cli\influxd-flags

Maybe an extra column?

### LINKED ISSUE #1728: TLS configuration missing for influxdb 2.0
_Describe the issue here._"
TLS settings and Recommended server configuration for 'modern compatibility' in InfluxDB 1.x docs isn't in 2.0 yet

##### Relevant URLs
1.x page: https://docs.influxdata.com/influxdb/v1.7/administration/config/#transport-layer-security-tls-settings
Equivalent 2.0 page: https://docs.influxdata.com/influxdb/v2.0/reference/config-options/

### PR DESCRIPTION
**Pull request by sanderson**
Closes #1726
Closes #1727
Closes #1728

- Removed table formatting for `influxd` flags and replaced them with links to each configuration option on the Configuration options docs.
- Added missing config options to the Configuration options doc (`influxql-*`, `storage-*`, and `tls-*`)

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

**Review by noramullen1 — APPROVED** [review]

File: content/influxdb/v2.0/reference/config-options.md [hunk 1]
```diff
@@ -366,8 +386,153 @@ http-bind-address = ":8086"
...
+[JSON](#)
+{{% /code-tabs %}}
+{{% code-tab-content %}}
+```yml
+iinfluxql-max-select-series: 0
+```
+{{% /code-tab-content %}}
+{{% code-tab-content %}}
+```toml
+iinfluxql-max-select-series = 0
```
Comment (pierwill) [inline content/influxdb/v2.0/reference/config-options.md:520]: influxql

File: content/influxdb/v2.0/reference/config-options.md [hunk 2]
```diff
@@ -366,8 +386,153 @@ http-bind-address = ":8086"
...
+
+### influxql-max-select-series
+Maximum number of series a `SELECT` statement can return.
+`0` allows an unlimited number of series.
+
+**Default:** `0`
+
+| influxd flag                    | Environment variable                 | Configuration key             |
+|:------------                    |:--------------------                 |:-----------------             |
+| `--iinfluxql-max-select-series` | `INFLUXD_INFLUXQL_MAX_SELECT_SERIES` | `iinfluxql-max-select-series` |
```
Comment (pierwill) [inline content/influxdb/v2.0/reference/config-options.md:494]: influxql

File: content/influxdb/v2.0/reference/config-options.md [hunk 3]
```diff
@@ -1127,18 +1333,802 @@ export INFLUXD_TLS_KEY=/path/to/influxdb.key
...
+}
+```
+{{% /code-tab-content %}}
+{{< /code-tabs-wrapper >}}
+
+---
+
+### storage-max-concurrent-compactions
+Maximum number of full and level compactions that can run concurrently.
+A value of `0` results in 50% of runtime. `GOMAXPROCS(0)` used at runtime.
```
Comment (pierwill) [inline content/influxdb/v2.0/reference/config-options.md:1404]: Should we be more specific about what "50% of runtime" means?

File: content/influxdb/v2.0/reference/config-options.md [hunk 4]
```diff
@@ -366,8 +386,153 @@ http-bind-address = ":8086"
...
+###### Configuration file
+{{< code-tabs-wrapper >}}
+{{% code-tabs %}}
+[YAML](#)
+[TOML](#)
+[JSON](#)
+{{% /code-tabs %}}
+{{% code-tab-content %}}
+```yml
+iinfluxql-max-select-series: 0
```
Comment (pierwill) [inline content/influxdb/v2.0/reference/config-options.md:515]: influxql

**Review by pierwill — APPROVED** [review]

**Review by russorat — APPROVED** [review]
LGTM ❤️

**Pull request merged**